python/visualization.py

```diff
@@ -988,27 +989,160 @@ def plot2D(
+                1,
+            ),
+        )
+
+        mesh.transform = transforms.MatrixTransform()
+        mesh.transform.translate(np.asarray(sim.geometry_center))
+        shading_filter = ShadingFilter(shininess=100)
+        mesh.attach(shading_filter)
+        view.add(mesh)
+
+    # Build source
+    thickness = (
+        sim.boundary_layers[0].thickness if not len(sim.boundary_layers) < 1 else 0
+    )
+    for source in sim.sources:
+        size = tuple(source.size)
+        source_box = Box(
+            *size,
+            color=(1, 0, 0, 1),  # red
+        )
+        center = list(source.center)
+        source_box.transform = transforms.MatrixTransform()
+        source_box.transform.translate(np.asarray(mesh_midpoint))
+        source_box.transform.translate(center)
+        source_box.transform.translate(tuple(sim.geometry_center))
+        view.add(source_box)
+
+    # Build monitors
+    for mon in sim.dft_objects:
+        for reg in mon.regions:
+            size = list(reg.size)
+            monitor_box = Box(
+                *size,
+                color=(0, 0, 1, 1),  # blue
+            )
+            center = list(reg.center)
+            monitor_box.transform = transforms.MatrixTransform()
+            vector = [0, 0, 0]
+            vector[reg.direction] = 1
+            vector = mp.Vector3(*vector)
+            monitor_box.transform.translate(tuple(mesh_midpoint))
+            monitor_box.transform.translate(center)
+            monitor_box.transform.translate(tuple(sim.geometry_center))
+            view.add(monitor_box)
+
+    # Build boundaries
+    for box_center_top in [
+        np.add(mesh_midpoint, (0, 0, sim_size[2] / 2 - thickness / 2)),
+        np.subtract(mesh_midpoint, (0, 0, sim_size[2] / 2 - thickness / 2)),
+    ]:
+        box = _build_3d_pml(sim_size[0], sim_size[1], thickness, box_center_top)
+        view.add(box)
+
+    for box_center_right in [
+        np.add(mesh_midpoint, (sim_size[0] / 2 - thickness / 2, 0, 0)),
+        np.subtract(mesh_midpoint, (sim_size[0] / 2 - thickness / 2, 0, 0)),
+    ]:
+        box = _build_3d_pml(thickness, sim_size[1], sim_size[2], box_center_right)
+        view.add(box)
+
+    for box_center_front in [
+        np.add(mesh_midpoint, (0, sim_size[1] / 2 - thickness / 2, 0)),
+        np.subtract(mesh_midpoint, (0, sim_size[1] / 2 - thickness / 2, 0)),
+    ]:
+        box = _build_3d_pml(sim_size[0], thickness, sim_size[2], box_center_front)
+        view.add(box)
+
+    def attach_headlight(view):
+        light_dir = (0, 1, 0, 0)
+        shading_filter.light_dir = light_dir[:3]
+        initial_light_dir = view.camera.transform.imap(light_dir)
+
+        @view.scene.transform.changed.connect
+        def on_transform_change(event):
+            transform = view.camera.transform
+            shading_filter.light_dir = transform.map(initial_light_dir)[:3]
+
+    attach_headlight(view)
+
+    # Plot
+    canvas.show(run=True)
```

Comment:
This opens an interactive window which is nice to play around with in order to set an appropriate viewing angle for the object. But how do we actually save the image to a file (without taking a screenshot)? The vispy menu does not seem to provide a `Save as image` feature.

Note that `plot2D` returns the axis handle to the plot which provides the option to the user to modify the plot in post processing or save it to a file:
https://github.com/NanoComp/meep/blob/d100be405292c1a3f33b05642a0333ac7c50db87/python/visualization.py#L984-L988

It would be good to follow a similar convention in `plot3D`. Also, perhaps we should update the short tutorial in `Basics.md` to demonstrate how to save an image to a file from the command line since it may not be obvious?